src/app/components/MediaLoader/index.test.tsx

```diff
@@ -168,4 +169,51 @@ describe('MediaLoader', () => {
       });
     });
   });
+
+  describe('Config', () => {
+    it('should use the pageIdentifierOverride when rendering On Demand TV', async () => {
+      const buildConfigSpy = jest.spyOn(buildConfig, 'default');
+      await act(async () => {
+        render(
+          <MediaPlayer
+            blocks={onDemandTvBlocks as MediaBlock[]}
+            pageIdentifierOverride="hindi.bbc_hindi_tv.tv.w172zm8b4tlpzxh.page"
+            embedded
+          />,
+          { service: 'hindi' },
+        );
+      });
+
+      expect(buildConfigSpy).toHaveBeenCalledWith(
+        expect.objectContaining({
+          counterName: 'hindi.bbc_hindi_tv.tv.w172zm8b4tlpzxh.page',
+        }),
+      );
+    });
+
+    it('should use the pageIdentifier from the EventTrackingContext', async () => {
+      const buildConfigSpy = jest.spyOn(buildConfig, 'default');
+      await act(async () => {
+        render(
+          <MediaPlayer blocks={onDemandTvBlocks as MediaBlock[]} embedded />,
+          {
+            service: 'hindi',
+            pageData: {
+              mediaBlocks: onDemandTvBlocks,
+              pageIdentifier: 'hindi.bbc_hindi_tv.tv_programmes.w13xttlw.page',
```

Comment:
I think the `pageIdentifier` should be set on the `atiData` rather than the `pageData` object and passed in as a param to the renderer. https://github.com/bbc/simorgh/blob/bb2b3a935f02ba359eb065cd3349bcfa093e9631/src/app/components/ATIAnalytics/params/genericPage/buildParams.ts#L23

Reply:
My apologies! Thanks for figuring that out, I only had a quick look and totally forgot about this mish-mash of migrated and un-migrated page types.